torch/_dynamo/variables/builder.py

```diff
@@ -873,11 +873,30 @@ def wrap_unspecialized_primitive(self, value):
 
                 shape_env = self.tx.output.shape_env
 
+                name = self.source.name()
+                if name not in self.tx.output.frame_state:
+                    curr_size = value
+                else:
+                    curr_size = self.tx.output.frame_state[name]
+                    if curr_size is not None and curr_size == value:
+                        pass
+                    else:
+                        curr_size = None
```

Comment:
I think we can shorten this as

<pre>
curr_size = frame_state.get(name, val)
if curr_size != val:
  curr_size = None
</pre>

Reply:
yes thats what I kept saying over VC lol. I'll rev it once I get CI signal that this is fine